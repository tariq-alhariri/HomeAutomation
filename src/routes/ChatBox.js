global.self = global;
import React from 'react';

import { View, Text, TouchableOpacity, TextInput, Alert, FlatList, StyleSheet, KeyboardAvoidingView, Image} from 'react-native';

import SocketIOClient from 'socket.io-client';
// import SocketIOClient from 'https://code.jquery.com/jquery-1.11.1.js'
import { Icon} from 'react-native-elements'; 


export default class ChatBox extends React.Component {

  static navigationOptions={
    header: null,    
    tabBarLabel:'ChatBox',
    tabBarIcon:({tintColor}) => (
      <Image 
        source={require('./icon/cha.png')}
        style={{width:24, height:24, tintColor:'white'}}
      >
      </Image>
    ) 
  }

  constructor(props){
    super(props)

    this.state={
      init:this.getCureentUser(),
        arr:[{user:'user', text:'please enter your message', date:'new Date'}], // initial values
        image:'',
        name:'',
        msg:''
      }

    this.socket = SocketIOClient('https://home99.herokuapp.com/');

    //fetch previous messages from the database
    this.socket.on('allDataBase',(data)=> {
      this.showArr(data);
    })

    //listen for mew messages
    this.socket.on('msg', (data)=> {
      this.showArr(data);
    })
  }

 //fetch user info
  async getCureentUser() {
    try {
      let response = await fetch('https://home99.herokuapp.com/user');
      let responseJson = await response.json();
      this.setState({image:responseJson.image})
      this.setState({name:responseJson.name})
    } 
    catch(error) {
      console.error(error);
    }
  } 

  // replace old array of messages with the newest one which includes new messages
  showArr(data){
    data = data.reverse();
    this.setState((pre)=> {
      return {
        arr:data
      }
    })
  }

  // save the text inside the Text Input in var msg
  changeMsg(data){
    this.setState({
      msg: data
    })
  }

  // send new message 
  send(data){
    this.socket.emit('message', {user: this.state.name, text: data, date:(new Date).toString()})
    this.setState({msg:''})
  }

  render(){
    return(
      <KeyboardAvoidingView behavior="padding" style={styles.container}>        
        <View>

          <TextInput
            placeholder = "Type here ..."
            onChangeText =  {(text) => this.changeMsg(text)}
            value = {this.state.msg}
          />

          <TouchableOpacity
            onPress={()=>{this.send(this.state.msg)}}
          >
            <Text>Send</Text>
          </TouchableOpacity>

          <FlatList
            data={this.state.arr}
            renderItem={({item}) =>
              <View>
                <Text style={styles.item}>
                  {item.user + ': '+item.text}
                </Text>
                <Text style={styles.items}> {item.date}</Text>
              </View>
            }
          />

        </View>  
      </KeyboardAvoidingView>
    )
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: 'rgb(135, 206, 250)',
    alignItems: 'center',
    justifyContent: 'center'
  },

  logoContainer: {
    alignItems: 'center',
    flexGrow: 1,
    justifyContent: 'center'
  },

  logo: {
    width: 200,
    height: 200
  },
  header: {
    color: '#fff',
    marginTop: 5,
    textAlign: 'center',
    opacity: 0.8,
    fontWeight: '700'
  },
  item: {
    padding: 10,
    fontSize: 18,
    height: 44,
  },
  items: {
    alignSelf:'flex-end',
    fontSize: 12,
  },
});