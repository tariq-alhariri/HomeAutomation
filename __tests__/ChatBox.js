import 'react-native';
import React from 'react';
import ChatBox from '../src/routes/ChatBox';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

test('Search snapshot test', () => {
  const component = renderer.create(<ChatBox />)
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot();
})


