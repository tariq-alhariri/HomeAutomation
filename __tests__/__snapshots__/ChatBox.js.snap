// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search snapshot test 1`] = `
<View
  keyboardVerticalOffset={0}
  onLayout={[Function]}
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "rgb(135, 206, 250)",
        "flex": 1,
        "justifyContent": "center",
      },
      Object {
        "paddingBottom": 0,
      },
    ]
  }
>
  <View>
    <TextInput
      onChangeText={[Function]}
      placeholder="Type here ..."
      value=""
    />
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
      >
        Send
      </Text>
    </View>
    <RCTScrollView
      data={
        Array [
          Object {
            "date": "new Date",
            "text": "please enter your message",
            "user": "user",
          },
        ]
      }
      disableVirtualization={false}
      getItem={[Function]}
      getItemCount={[Function]}
      horizontal={false}
      initialNumToRender={10}
      keyExtractor={[Function]}
      maxToRenderPerBatch={10}
      numColumns={1}
      onContentSizeChange={[Function]}
      onEndReachedThreshold={2}
      onLayout={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      onViewableItemsChanged={undefined}
      renderItem={[Function]}
      scrollEventThrottle={50}
      stickyHeaderIndices={Array []}
      updateCellsBatchingPeriod={50}
      windowSize={21}
    >
      <View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View>
            <Text
              accessible={true}
              allowFontScaling={true}
              disabled={false}
              ellipsizeMode="tail"
              style={
                Object {
                  "fontSize": 18,
                  "height": 44,
                  "padding": 10,
                }
              }
            >
              user: please enter your message
            </Text>
            <Text
              accessible={true}
              allowFontScaling={true}
              disabled={false}
              ellipsizeMode="tail"
              style={
                Object {
                  "alignSelf": "flex-end",
                  "fontSize": 12,
                }
              }
            >
               
              new Date
            </Text>
          </View>
        </View>
      </View>
    </RCTScrollView>
  </View>
</View>
`;
